src/server/hook_func.c

```diff
@@ -6471,22 +6471,25 @@ add_pending_mom_allhooks_action(void *minfo, unsigned int action)
 void
 next_sync_mom_hookfiles(void)
 {
-	int	timeout_sec;
+	long	timeout_sec;
 	time_t	timeout_time;
 	time_t	current_time;
+	int	timed_out = 0;
 
 	if (pbs_conf.pbs_use_tcp == 1) {
-		timeout_sec = SYNC_MOM_HOOKFILES_TIMEOUT_TPP;
+		if (server.sv_attr[(int)SRV_ATR_sync_mom_hookfiles_timeout].at_flags & ATR_VFLAG_SET) {
+			timeout_sec = server.sv_attr[(int)SRV_ATR_sync_mom_hookfiles_timeout].at_val.at_long;
+		} else {
+			timeout_sec = SYNC_MOM_HOOKFILES_TIMEOUT_TPP;
+		}
```

Comment:
Initialize it as had been done, then check whether the values is set and update it. Reduces code to three lines.